src/darwin/Framework/CHIP/MTROperationalCredentialsDelegate.mm

```diff
@@ -178,12 +178,10 @@
     chip::ByteSpan firmwareInfoSpan;
     chip::Credentials::DeviceAttestationVendorReservedDeconstructor vendorReserved;
 
-    __block chip::Optional<chip::Controller::CommissioningParameters> commissioningParameters;
     // Dereferencing mCppCommissioner as it would be set to point to a valid Cpp commissioner by now, as we are in the middle of
     // commissioning
-    dispatch_sync(mChipWorkQueue, ^{
-        commissioningParameters = mCppCommissioner->GetCommissioningParameters();
-    });
+    chip::Optional<chip::Controller::CommissioningParameters> commissioningParameters
+        = mCppCommissioner->GetCommissioningParameters();
```

Comment:
Again, none of this is addressing mCppCommissioner dangling.

At bare minimum when the MTRDeviceController shuts down it should call back into this code and null out `mCppCommissioner`, and we should have null-checks guarding use of it.

Perhaps better would be to not store mCppCommissioner at all, expose non-public APIs on MTRDeviceController to do whatever we need to do here, which do `checkIsRunning`, etc, hold on to an MTRDeviceController in this object, and make sure that controller shutdown nulls out that ref when it shuts down to avoid it leaking.  But that still involves the controller calling into this class when it shuts down, so it might be simpler to just null out `mCppCommissioner` at that point.

Reply:
I followed the first suggestion (i.e. "when the MTRDeviceController shuts down it should call back into this code and null out mCppCommissioner, and we should have null-checks guarding use of it.") 

Code updated.